src/components/back-to-top.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useEffect, useState } from "react";
+import { FaCaretUp } from "react-icons/fa";
+
+const BackToTop: React.FC<{}> = () => {
+  const [visible, setVisible] = useState(false);
+
+  const listener: EventListener = () => {
+    if (window.scrollY > 0) {
+      setVisible(true);
+    } else {
+      setVisible(false);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener("scroll", listener);
+    return (): void => {
+      window.removeEventListener("scroll", listener);
+    };
+  });
```

Comment:
not sure how this functions get executed when `scrollToTop` function is called. Mind giving a TLDR?

Reply:
The event listener is just to update the React state var `visible`. Not executed in `scrollToTop`.

useEffect here is a substitute of traditional class based React `componentDidMount`.